src/lsm/tree.zig

```diff
@@ -51,6 +51,21 @@ const half_bar_beat_count = @divExact(config.lsm_batch_multiple, 2);
 /// The maximum number of tables for a single tree.
 pub const table_count_max = table_count_max_for_tree(config.lsm_growth_factor, config.lsm_levels);
 
+/// The upper-bound count of input tables to a single tree's compaction.
+///
+/// +1: from level A
+/// +1: from level B, with one edge overlapping the lsm_growth_factor tables.
+///     (This is not +2 because input table selection is least-overlap. If the input table overlaps
+///     on both edges, there must be another table with less overlap to select).
+pub const compaction_tables_input_max = 1 + config.lsm_growth_factor + 1;
```

Comment:
I'm still confused about why this is +1 and not +0. 

Suppose every table in level A overlaps at least growth_factor + 1 tables in level B.
Assign each table in level B to the first table in level A that overlaps it.
Then the first table in level A has at least growth_factor + 1 tables assigned to it (because it gets the first overlap), and every other table in level A has at least growth_factor tables assigned.
So the total number of tables in level B must be at least (level_a_table_count * growth_factor) + 1, which is too many.

Reply:
You're correct; I just didn't understand that before :+1:.